middle_end/flambda/simplify/expr_builder.ml

```diff
@@ -110,7 +111,7 @@ let create_let uacc (bound_vars : BLB.t) defining_expr
         not (has_uses || (generate_phantom_lets && can_phantomise))
       in
       if will_delete_binding then begin
-        bound_vars, None, Defining_expr_deleted_at_runtime
+        bound_vars, None, Defining_expr_deleted_at_compile_time
```

Comment:
I wonder if there might be a pre-existing bug below on line 126: if the binding was already phantom, upon entry to this function, I think we might say "defining expr deleted at runtime" even though it was already deleted.  Please leave a CR to check.  (This is why we need to test with more than one round of Simplify, although in this case it would be hard to spot anyway.)

Reply:
I think you're exactly right, actually—the preexisting code that inspects this value will double-count phantomized bindings whenever there's a second pass. Added a CR.